xml/src/main/java/org/ehcache/xml/ServiceConfigurationParser.java

```diff
@@ -55,9 +55,9 @@
     new DefaultCacheEventListenerConfigurationParser()
   );
 
-  private final Set<CacheServiceConfigurationParser<?>> extensionParsers;
+  private final Collection<CacheServiceConfigurationParser<?>> extensionParsers;
 
-  public ServiceConfigurationParser(Set<CacheServiceConfigurationParser<?>> extensionParsers) {
+  public ServiceConfigurationParser(Collection<CacheServiceConfigurationParser<?>> extensionParsers) {
```

Comment:
Instead of passing the collection of parsers in here and re-generating a map out of it, the maps itself that's created in `ConfigurationParser` could have been passed in.